rwengine/src/data/GameTexts.hpp

```diff
@@ -3,21 +3,42 @@
 #include <string>
 #include <unordered_map>
 
+/**
+ * The game stores strings as 16-bit indexes into the font
+ * texture, which is something simllar to ASCII.
+ */
+using GameString = std::basic_string<char16_t>;
```

Comment:
This should be `uint16_t` rather than `char16_t`.

We can't use std::string as they only store single byte chars. fromString should only be used for the odd place where we need to convert some ASCII alphanumerics to a `GameString`, such as in the formatting of numbers etc.

The 8 bit chars for keys are already handled by `GameStringKey`. I hadn't considered storing the keys as 64-bit integers, since all keys are 8 bytes we should be able to get away with that unless there is some junk after each key (I'd hope not). I can look into that to see if it makes things a bit simpler.